core/services/ardupilot_manager/firmware_install/FirmwareInstall.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import pathlib
+import shutil
+import stat
+
+from loguru import logger
+
+from firmware_download.FirmwareDownload import Platform
+from firmware_upload.FirmwareUpload import FirmwareUpload
+
+
+class InvalidFirmwareFile(ValueError):
+    pass
+
+
+class UnsupportedPlatform(ValueError):
+    pass
+
+
+class FirmwareInstallFail(RuntimeError):
+    pass
+
+
+class FirmwareInstaller:
+    def __init__(self, firmware_folder: pathlib.Path) -> None:
+        self._firmware_folder: pathlib.Path = firmware_folder
+
+    def set_platform(self, platform: Platform) -> None:
+        self._platform = platform
+
+    @property
+    def firmware_name(self) -> str:
+        return f"ardupilot_{self._platform.value.lower()}"
+
+    @property
+    def firmware_path(self) -> pathlib.Path:
+        return pathlib.Path.joinpath(self._firmware_folder, self.firmware_name)
+
+    def is_firmware_installed(self) -> bool:
+        if self._platform == Platform.Pixhawk1:
+            # Assumes for now that a Pixhawk always has a firmware installed, which is true most of the time
+            return True
+
+        if self._platform in [Platform.Navigator, Platform.SITL]:
+            return pathlib.Path.is_file(self.firmware_path)
+
+        logger.error("Platform not supported.")
+        raise UnsupportedPlatform
+
+    def is_valid_firmware(self, firmware_path: pathlib.Path) -> bool:
+        if self._platform == Platform.Pixhawk1:
+            # For Pixhawk/Serial boards only .apj files are supported
```

Comment:
@Williangalvani, @rafaellehmkuhl  can you check if ardupilot always provides an apj file for custom or local build firmwares ?
Otherwise we'll need to support normal binary files for serial boards.